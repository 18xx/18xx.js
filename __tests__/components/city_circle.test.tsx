import CityCircle from '../../src/components/city_circle';

import Point from '../../src/point';

describe('CityCircle', () => {
  describe('.RADIUS', () => {
    it('returns the value of the RADIUS const', () => {
      expect(CityCircle.RADIUS).toEqual(19);
    });
  });

  describe('.STROKE_WIDTH', () => {
    it('returns the value of the STROKE_WIDTH const', () => {
      expect(CityCircle.STROKE_WIDTH).toEqual(2);
    });
  });

  describe('#toString()', () => {
    it('returns an svg representation of the circle', () => {
      const point: Point = new Point(50, 40);
      const cc: CityCircle = new CityCircle({ point });
      expect(cc.toString()).toMatchSnapshot();
    });
  });
});
