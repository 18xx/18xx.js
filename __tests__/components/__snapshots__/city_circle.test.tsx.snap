exports[`CityCircle #toString() returns an svg representation of the circle 1`] = `"[object Object]"`;
